### PR TITLE
docs for deepsparse.Pipelines and update server readme

### PR DESCRIPTION
**Pull request by bfineran**
simple documentation/quickstart for `deepsparse.Pipeline`. Thoughts on other functionality to include would be appreciated. Deciding if more information on configs should be added.

`server/README.md` updated to include new pipelines info

### CODE REVIEW

File: PIPELINES.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+Custom tasks may be defined by inheriting from the base Pipeline class in 
+[`pipelines.py`](https://github.com/neuralmagic/deepsparse/blob/main/src/deepsparse/pipeline.py)
+and implementing the `abstractmethod` stubs.  This includes defining input and output
+pydantic schemas and defining pre and post processing flows.
+
+Tasks may also be registered to `Pipeline.create` by decorating the implemented class
+with `Pipeline.register`. This enables easy instantiation of the new pipeline in existing
+scripts and flows by refering to the new task name.
+
+#### `Pipeline.register` Example
```
Comment (mgoin) [inline PIPELINES.md:69]:
```suggestion
with `@Pipeline.register`. This enables easy instantiation of the new pipeline in existing
scripts and flows by referring to the new task name.

#### `@Pipeline.register` Example
```

**Review by mgoin — COMMENTED** [review]
Looks good with a skim. Will sweep through again and accept

File: src/deepsparse/server/README.md [hunk 2]
```diff
@@ -4,7 +4,10 @@
 pip install deepsparse[server]
 ```
 
-The DeepSparse inference server allows you to serve models and pipelines for deployment in HTTP. The server runs on top of the popular FastAPI web framework and Uvicorn web server. Currently, the server only supports NLP tasks; however, support for computer vision will soon be released in upcoming versions.
+The DeepSparse inference server allows you to serve models and pipelines for deployment in HTTP. The server runs on top of the popular FastAPI web framework and Uvicorn web server.
+The server supports any task from deepsparse.Pipeline including NLP, classification, and detection tasks.
```
Comment (dbogunowicz) [inline src/deepsparse/server/README.md:8]:
```suggestion
The server supports any task from deepsparse.Pipeline including NLP, object classification, and object detection tasks.
```
Nit: being very explicit.

Comment (bfineran) [inline src/deepsparse/server/README.md:8]:
```suggestion
The server supports any task from deepsparse. Pipeline including NLP, image classification, and object detection tasks.
```

File: PIPELINES.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## Supported Tasks
+
+Development of new Pipelines for tasks is always ongoing. Currently supported tasks include:
+
+| Domain |         Task         | Documentation |
+|--------|:--------------------:|:-------------:|
+| NLP    | text-classification  |      TBA      |
+| NLP    | token-classification |      TBA      |
+| NLP    |  question-answering  |      TBA      |
```
[the remark this answers is not in the document]
Comment (bfineran) [reply PIPELINES.md:44]: we don't have a pipeline for it yet which is why it is not here. it is used to train upstreams, but there are still some downstream use cases (ie we can use it for IR or the zero shot pipeline michael and kyle are working through)

**Review by dbogunowicz — APPROVED** [review]
looks good to me!

**Review by mgoin — APPROVED** [review]

**Pull request merged**